streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBVersionedStore.java

```diff
@@ -266,75 +269,28 @@ public VersionedRecord<byte[]> get(final Bytes key, final long asOfTimestamp) {
         return null;
     }
 
-    public VersionedRecordIterator<byte[]> get(final Bytes key, final long fromTimestamp, final long toTimestamp, final boolean isAscending) {
-
-        Objects.requireNonNull(key, "key cannot be null");
+    @SuppressWarnings("unchecked")
```

Comment:
> Why do we need this suppression?

Because of the following warning when creating a `LogicalSegmentIterator` object ([line# 280](https://github.com/aliehsaeedii/kafka/blob/30f8d197319359074924b25041f63387484dbc08/streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBVersionedStore.java#L280), and [line# 293](https://github.com/aliehsaeedii/kafka/blob/30f8d197319359074924b25041f63387484dbc08/streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBVersionedStore.java#L293))
`Unchecked assignment: 'org.apache.kafka.streams.state.internals.LogicalSegmentIterator' to 'org.apache.kafka.streams.state.VersionedRecordIterator<byte[]>' `

Reply:
Well, `VersionedRecordIterator<V>` takes generic `V` and `LogicalSegmentIterator ` is not specifying the type (what is a bug):
```
public class LogicalSegmentIterator implements VersionedRecordIterator {
```

Guess it should be, what will avoid the warning making the suppression. unnecessary?
```
public class LogicalSegmentIterator implements VersionedRecordIterator<byte[]> {
```

What implies that `public Object next() ` should actually return `byte[]`, right (something I missed in my review).